x-pack/metricbeat/module/googlecloud/stackdriver/metricset.go

```diff
@@ -0,0 +1,168 @@
+	ExcludeLabels       bool     `config:"exclude_labels"`
+	ServiceName         string   `config:"stackdriver.service"  validate:"required"`
+	CredentialsFilePath string   `config:"credentials_file_path"`
+
+	credentialsOption option.ClientOption
+	opt               []option.ClientOption
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Beta("The gcp '%s' metricset is beta.", MetricsetName)
+
+	m := &MetricSet{BaseMetricSet: base}
+
+	if err := base.Module().UnpackConfig(&m.config); err != nil {
+		return nil, err
+	}
+
+	m.config.credentialsOption = option.WithCredentialsFile(m.config.CredentialsFilePath)
+
+	if err := validatePeriodForGCP(m.Module().Config().Period); err != nil {
+		return nil, err
+	}
+
+	return m, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right
+// format. It publishes the event which is then forwarded to the output. In case
+// of an error set the Error field of mb.Event or simply call report.Error().
+func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) (err error) {
+	httpClient, err := createHTTPClient(ctx, m.config.opt...)
+	if err != nil {
+		return errors.Wrapf(err, "could not create HTTP client to connect to Google Services")
+	}
+
+	httpClientOption := option.WithHTTPClient(httpClient)
+	m.config.opt = []option.ClientOption{m.config.credentialsOption, httpClientOption}
+
+	reqs, err := newStackdriverMetricsRequester(ctx, m.config, m.Module().Config().Period, m.Logger())
+	if err != nil {
+		return errors.Wrapf(err, "error trying to do create a request client to GCP project '%s' in zone '%s'", m.config.ProjectID, m.config.Zone)
+	}
+
+	responses, err := reqs.Metrics(ctx, m.config.Metrics)
+	if err != nil {
+		return errors.Wrapf(err, "error trying to get metrics for project '%s' and zone '%s'", m.config.ProjectID, m.config.Zone)
+	}
+
+	events, err := m.eventMapping(ctx, responses)
+	if err != nil {
+		return err
+	}
+
+	for _, event := range events {
+		reporter.Event(event)
+	}
+
+	return nil
+}
+
+//createHTTPClient to use with share around every request to GCP services. It's recreated in every Fetch call.
+func createHTTPClient(ctx context.Context, opts ...option.ClientOption) (*http.Client, error) {
+	transport, err := httpGoogle.NewTransport(ctx, http.DefaultTransport, opts...)
+	if err != nil {
+		return nil, err
+	}
+	return &http.Client{
+		Transport: transport,
+		Timeout:   30 * time.Second,
+	}, nil
+}
+
+func (m *MetricSet) eventMapping(ctx context.Context, tss []*monitoringpb.TimeSeries) ([]mb.Event, error) {
+	e := newIncomingFieldExtractor(m.Logger())
+
+	var gcpService = googlecloud.NewStackdriverMetadataServiceForTimeSeries(nil)
+	var err error
+	// Override default metadata service labels are ON and service is known
+	if m.config.ExcludeLabels {
```

Comment:
I guess this will need to be negated after the change from `IncludeLabels` to `ExcludeLabels`.
```suggestion
	if !m.config.ExcludeLabels {
```

Reply:
Good catch :smile: